besu/src/main/java/org/hyperledger/besu/cli/options/unstable/ChainDataPruningOptions.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package org.hyperledger.besu.cli.options.unstable;
+
+import org.hyperledger.besu.util.number.PositiveNumber;
+
+import picocli.CommandLine;
+
+public class ChainDataPruningOptions {
+
+  public static final long DEFAULT_CHAIN_DATA_PRUNING_MIN_BLOCKS_RETAINED = 50400;
+  public static final int DEFAULT_CHAIN_DATA_PRUNING_FREQUENCY = 256;
+
+  @CommandLine.Option(
+      hidden = true,
+      names = {"--Xchain-data-pruning-enabled"},
```

Comment:
Yea, the reason for having the data here is to emphasize that all data related to the chain will be pruned. For example, besu stores transaction location as part of a block import, which isn't actually part of the chain but is a piece of data related to the chain.

Reply:
I'd argue transaction location is an implementation detail we shouldn't encode into this name and so data is still redundant.
 
I think there's an important distinction between this pruning vs world state pruning and I'm not sure the average user would understand that from "chain-data". One may assume world state is part of the "chain".

What are your thoughts on "block-data"?

Reply:
That's true also. I'd say it is better to rename the existing pruning to something like "state-pruning" once this becomes a visible option to not confuse users. 

I would prefer "chain-pruning"/"chain-data-pruning" over "block-data-pruning" as we are pruning the chain by removing old blocks from it. The latter may create the impression that we are pruning some data from blocks. 

Wondering about the thoughts of others regarding the naming @jframe @usmansaleem @gfukushima

Reply:
That's true. Maybe "block-pruning" is more accurate than "block-data-pruning". 
It hides the details about pruning other data associated with the block but I think that's fine.

I'd also be fine with "chain-pruning". 
Confusion with state pruning probably isn't that big a deal, especially since the users of this will be on bonsai so shouldn't be configuring state pruning (although some people still try!)

Reply:
Cool, just updated